webapp/apps/btax/helpers.py

```diff
@@ -0,0 +1,262 @@
+                'mpl_toolkits.mplot3d', 'pandas']
+sys.modules.update((mod_name, Mock()) for mod_name in MOCK_MODULES)
+
+PYTHON_MAJOR_VERSION = sys.version_info.major
+
+BTAX_BITR = ['btax_betr_corp',
+             'btax_betr_pass',
+             'btax_betr_entity_Switch']
+BTAX_DEPREC = ['btax_depr_allyr', 'btax_depr_3yr', 'btax_depr_5yr',
+               'btax_depr_7yr', 'btax_depr_10yr',
+               'btax_depr_15yr','btax_depr_20yr', 'btax_depr_25yr',
+               'btax_depr_27_5yr', 'btax_depr_39yr']
+BTAX_OTHER = ['btax_other_hair', 'btax_other_corpeq',
+              'btax_other_proptx','btax_other_invest']
+BTAX_ECON = ['btax_econ_nomint', 'btax_econ_inflat',]
+
+
+BTAX_VERSION_INFO = btax._version.get_versions()
+BTAX_VERSION = ".".join([BTAX_VERSION_INFO['version'], BTAX_VERSION_INFO['full-revisionid'][:6]])
+
+#
+# Prepare user params to send to DropQ/Taxcalc
+#
+class BTaxField(TaxCalcField):
+
+    pass
+
+
+class BTaxParam(object):
+
+    """
+    A collection of TaxCalcFields that represents all configurable details
+    for one of TaxCalc's Parameters
+    """
+    coming_soon = False
+    inflatable = False
+    def __init__(self, param_id, attributes):
+        self.__load_from_json(param_id, attributes)
+
+    def __load_from_json(self, param_id, attributes):
+        # TODO does /btax need to handle
+        # budget year /start year logic
+        # as in /taxbrain. If so, see
+        # TaxCalcParam for changes to
+        # make here
+        self.start_year = first_budget_year = 2015
+
+        values_by_year = attributes['value']
+        col_labels = attributes['col_label']
+
+        self.tc_id = param_id
+        self.nice_id = param_id[1:] if param_id[0] == '_' else param_id
+        self.name = attributes['long_name']
+        self.info = " ".join([
+            attributes['description'],
+            attributes.get('notes') or ""     # sometimes this is blank
+            ]).strip()
+
+        # normalize single-year default lists [] to [[]]
+        if not isinstance(values_by_year[0], list):
+            values_by_year = [values_by_year]
+
+        # organize defaults by column [[A1,B1],[A2,B2]] to [[A1,A2],[B1,B2]]
+        values_by_col = [list(x) for x in zip(*values_by_year)]
+
+        # create col params
+        self.col_fields = []
+
+        self.col_fields.append(TaxCalcField(
+            self.nice_id,
+            attributes['description'],
+            values_by_col[0],
+            self,
+            first_budget_year
+        ))
+
+        # we assume we can CPI inflate if first value isn't a ratio
+        first_value = self.col_fields[0].values[0]
+
+        validations_json =  attributes.get('validations')
+        if validations_json:
+            self.max = validations_json.get('max')
+            self.min = validations_json.get('min')
+        else:
+            self.max = None
+            self.min = None
+
+        # Coax string-formatted numerics to floats and field IDs to nice IDs
```

Comment:
I'll go over the max / min validation here.  It appears to be looking in the wrong dict for max / min (which are in the `btax/param_defaults/btax_defaults.json`)